### PR TITLE
feat(ng6): initial migration to ng-cli6 

### PR DESCRIPTION
**Pull request by invincibleJai**
### Todo list:

- [x] migrate editor library to ng6  and use ng-cli6 
- [x] CI/CD to run unit test for library
- [x] migrate demo app to ng6  and use ng-cli6 
- [x] support env variable for demo app.
- [x] update README instructions for new dev environment 
- [x] test demo app and make sure it works as intended
- [x] make sure all test cases pass
- [x] make sure npm release content all required file
- [ ] release version `v1.0.0` with semantic-release

### CODE REVIEW

**Comment by joshuawilson** [comment]
I didn't realize you were going to do this upgrade. I made a traditional one on another PR. Let me know if you want me to finish it.

**Comment by invincibleJai** [comment]
@joshuawilson yes we started with it and @corinnekrych is helping/guiding me with this.

Will let you know if we need your help with it.

File: package.json [hunk 1]
```diff
@@ -1,203 +1,83 @@
...
-    "rxjs": "^5.5.2"
+    "core-js": "^2.5.4",
+    "ngx-base": "^3.1.1",
+    "ngx-bootstrap": "^3.0.1",
+    "patternfly": "^3.54.1",
+    "ngx-fabric8-wit": "^7.1.0",
+    "ngx-login-client": "^2.3.1",
+    "ngx-modal": "^0.0.29",
+    "rxjs": "^6.0.0",
+    "rxjs-compat": "^6.3.0",
```
Comment (corinnekrych) [inline package.json]: @invincibleJai I think this dependency should be removed as we now use rxjs6 for all dependencies (latest fabric8-wit) on both `package.json`

**Comment by corinnekrych** [comment]
@invincibleJai two mains comments:
- let's remove rxjs-compat dep
- could we remove the shared folder in demo app to simplify the code since we may not need env variable (url are read form param)

**Comment by invincibleJai** [comment]
@jyasveer pls verify last two commits , if all looks good to you. Thanks

**Pull request merged**